bluepill/src/BPPacker.m

```diff
@@ -129,27 +165,31 @@ @implementation BPPacker
         if (config.testCasesToSkip && [config.testCasesToSkip count] > 0) {
             [bundleTestsToRun minusSet:[[NSSet alloc] initWithArray:config.testCasesToSkip]];
         }
+        [BPUtils printInfo:INFO withString:@"Bundle: %@; All Tests count: %lu; bundleTestsToRun count: %lu", xctFile.testBundlePath, (unsigned long)[xctFile.allTestCases count], (unsigned long)[bundleTestsToRun count]];
         if (bundleTestsToRun.count > 0) {
+            testsToRunByFilePath[xctFile.testBundlePath] = bundleTestsToRun;
             double __block testBundleExecutionTime = 0.0;
             [bundleTestsToRun enumerateObjectsUsingBlock:^(id _Nonnull test, BOOL * _Nonnull stop) {
                 // TODO: Assign a sensible default if the estimate is not given
                 if ([testTimes objectForKey:test]) {
                     testBundleExecutionTime += [[testTimes objectForKey:test] doubleValue];
                 }
             }];
-            estimatedTimesByTestFilePath[xctFile.testBundlePath] = [NSNumber numberWithDouble:testBundleExecutionTime];
             totalTime += testBundleExecutionTime;
+            testEstimatesByFilePath[xctFile.testBundlePath] = [NSNumber numberWithDouble:testBundleExecutionTime];
         }
     }
-    assert([estimatedTimesByTestFilePath count] == [xcTestFiles count]);
+    assert([testEstimatesByFilePath count] == [xcTestFiles count]);
+    
+    double optimalBundleTime = MAX(1, totalTime / MAX (1, [[config numSims] integerValue]));
```

Comment:
Why these `MAX(1, ` calls? The second one doesn't make sense to me... have you ever seen `numSims` be _less than_ one?

Also it's not clear to me that this is an _optimal_ bundle time. Say we have 10 bundles and 4 sims, total run time of all tests is 20 min. This would set the optimal time to 5 min. But with 10 bundles at 5 min you'd get 50 min. 

I think you want something like total time divided by numSims _times_ numBundles no? 

Also this doesn't seem to be taking into account bundles that can't be split.

Reply:
I'll add more documentation but the idea is... if all bundles take a total of 20 minutes using 4 sims, then it would be better if none of the bundles take longer than 20/4=5 minutes. 

For example... if 10 bundles take 10, 2, 2, 1, 1, 1, 1, 1, 0.5 and 0.5 with 4 num-sims then the optimal bundle time will come out to be 5 minutes and they shall be split into...
5, 5, 2, 2, 1, 1, 1, 1, 1, 0.5 and 0.5 and shall execute as below...
Sim#1: 5
Sim#2: 5
Sim#3: 2, 1, 1, 1
Sim#4: 2, 1, 1, 0.5, 0.5
which will make the long pole in this case to be 5 minutes.

Reply:
I think the confusion stems from our terminology. I wonder if we should rename the `numSims` variable to `numTracks` and start using your idea of having _parallel tracks_ and simulators that run in those tracks.